src/shared/sleep-config.c

```diff
@@ -247,20 +335,17 @@ int get_battery_discharge_rate(void) {
 }
 
 /* Write battery percentage discharge rate per hour along with system and battery identifier hash to file */
-int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
-        uint64_t system_hash_id;
+int put_battery_discharge_rate(int estimated_battery_discharge_rate, uint64_t system_hash_id) {
         int r;
 
+        assert(system_hash_id);
+
         if (!battery_discharge_rate_is_valid(estimated_battery_discharge_rate))
                return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
 
-        r = get_system_battery_identifier_hash(&system_hash_id);
-        if (r < 0)
-               return log_debug_errno(r, "Failed to generate system battery identifier hash: %m");
-
         r = write_string_filef(
                         DISCHARGE_RATE_FILEPATH,
-                        WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_MKDIR_0755,
+                        WRITE_STRING_FILE_CREATE | WRITE_STRING_FILE_MKDIR_0755 | WRITE_STRING_FILE_TRUNCATE,
```

Comment:
this will truncate every time, so only the last battery will be saved - add a `bool truncate` parameter, pass it only on the first iteration of the loop, and add the flag only if it's true

Reply:
ok. since i have not tested the code even once because the build did not pass on local, misunderstood the working. this would still need test and fix. Except this one, i have addressed all the review comments.

Reply:
this is still pending